### PR TITLE
refactor: Improve pubspecs files and example projects

### PR DESCRIPTION
**Pull request by mbfakourii**
## Pull Request

- Report security issues [confidentially](https://github.com/parse-community/Parse-SDK-Flutter/security/policy).
- Any contribution is under this [license](https://github.com/parse-community/Parse-SDK-Flutter/blob/master/LICENSE).
- Link this pull request to an [issue](https://github.com/parse-community/Parse-SDK-Flutter/issues?q=is%3Aissue).

## Issue
Closes: #957

## Approach
Improved pubspecs files base [dart site](https://dart.dev/tools/pub/pubspec) and example projects

## Tasks

- [ ] Add tests
- [ ] Add changes to documentation (guides, repository pages, code comments)

### CODE REVIEW

**Review by mtrezza — APPROVED** [review]
Looks good! Is this ready for merge?

**Comment by mbfakourii** [comment]
> Looks good! Is this ready for merge?

Yes, it is ready

**Comment by mbfakourii** [comment]
this fail test for [this bug](https://github.com/parse-community/Parse-SDK-Flutter/issues/958)

**Comment by mtrezza** [comment]
If we restart the CI it should work, since dio has fixed the issue and there's no lock file.

**Comment by mbfakourii** [comment]
> If we restart the CI it should work, since dio has fixed the issue and there's no lock file.

What is your suggestion to improve CI?

**Comment by mtrezza** [comment]
Improve the CI?

**Comment by mbfakourii** [comment]
> Improve the CI? What do you mean?

I meant [here](https://github.com/parse-community/Parse-SDK-Flutter/pull/959#issuecomment-1666502310)

**Comment by mtrezza** [comment]
I don't think there's anything to improve. In that comment I was referring to your [comment](https://github.com/parse-community/Parse-SDK-Flutter/pull/956#issuecomment-1661974717) where you said that the in this PR here fails because of the dio issue. Since the CI now passes without us changing the dio dependency version, we know that the issue self-resolved, as expected without us having to publish a release.

File: packages/dart/pubspec.yaml [hunk 1]
```diff
@@ -1,7 +1,21 @@
...
+funding:
+  - https://opencollective.com/parse-server
+  - https://github.com/sponsors/parse-community
+
+topics:
+  - parse
+  - parse-platform
+  - SDK
+  - node-js
+  - back-end
```
Comment (mtrezza) [inline packages/dart/pubspec.yaml:18]:
```suggestion
  - parse
  - parse-platform
  - parse-server
  - SDK
  - node-js
  - backend
```

File: packages/flutter/pubspec.yaml [hunk 2]
```diff
@@ -1,7 +1,21 @@
...
+funding:
+  - https://opencollective.com/parse-server
+  - https://github.com/sponsors/parse-community
+
+topics:
+  - parse
+  - parse-platform
+  - SDK
+  - node-js
+  - back-end
```
Comment (mtrezza) [inline packages/flutter/pubspec.yaml:18]:
```suggestion
  - parse
  - parse-platform
  - parse-server
  - SDK
  - node-js
  - backend
```

**Review by mtrezza — DISMISSED** [review]
Looks good! Could you resolve the conflicts, when this seems ready to merge.

**Review by mtrezza — APPROVED** [review]

**Comment by mbfakourii** [comment]
> Looks good! Is this ready to merge?

[`Tag each package with at most 5 topics`](https://dart.dev/tools/pub/pubspec#topics)

File: packages/flutter/pubspec.yaml [hunk 3]
```diff
@@ -1,7 +1,22 @@
...
+  - https://opencollective.com/parse-server
+  - https://github.com/sponsors/parse-community
+
+topics:
+  - parse
+  - parse-platform
+  - parse-server
+  - SDK
+  - node-js
+  - backend
```
Comment (mtrezza) [inline packages/flutter/pubspec.yaml:19]:
```suggestion
  - parse
  - parse-platform
  - parse-server
  - node-js
  - backend
```

**Review by mtrezza — APPROVED** [review]
How about now?

**Comment by mbfakourii** [comment]
> How about now?

It looks good.

**Comment by mbfakourii** [comment]
> How about now?

It looks good

**Pull request merged**